src/Processors/Executors/PipelineExecutor.cpp

```diff
@@ -46,6 +44,16 @@ PipelineExecutor::PipelineExecutor(Processors & processors, QueryStatus * elem)
 
         throw;
     }
+    if (process_list_element)
+    {
+        auto settings = process_list_element->context.lock()->getSettings();
```

Comment:
```suggestion
        auto settings = process_list_element->getContext()->getSettings();
```

And maybe clang tidy will be fixed by this?